include/nbl/builtin/hlsl/ieee754.hlsl

```diff
@@ -152,6 +152,12 @@ NBL_CONSTEXPR_INLINE_FUNC FloatingPoint flipSign(FloatingPoint val)
 	return bit_cast<FloatingPoint>(asUint ^ ieee754::traits<AsFloat>::signMask);
 }
 
+template <typename FloatingPoint NBL_FUNC_REQUIRES(concepts::FloatingPointLikeScalar<FloatingPoint>)
+NBL_CONSTEXPR_INLINE_FUNC FloatingPoint flipSign(FloatingPoint val, bool flip)
+{
+	return flip ? flipSign(val) : val;
```

Comment:
its more efficient if you do `bit_Cast<FloatingPoint>(asUint^(flip ? SignMask:ZeroMask))` also then you don't need two separate functions, whenever the bool is compile time known up-front the compiler will constant fold the mask